src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -480,31 +501,128 @@ public override async Task RunAsync()
             await runTask;
         }
 
+        private async Task<bool> ShouldStartChildProcessFromHostRuntime(bool isVerbose)
+        {
+            if (string.Equals(HostRuntime, "default", StringComparison.OrdinalIgnoreCase))
+            {
+                if (isVerbose)
+                {
+                    PrintVerboseForHostSelection(isVerbose, "out-of-process");
+                }
+                return false;
+            }
+            else if (string.Equals(HostRuntime, DotnetConstants.InProc8HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (await ShouldNet8InProcBeLaunched())
+                {
+                    if (isVerbose)
+                    {
+                        PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc8HostRuntime);
+                    }
+                    return true;
+                }
+                else
+                {
+                    throw new CliException($"Invalid config for running {DotnetConstants.InProc8HostRuntime} host.");
+                }
+            }
+            else if (string.Equals(HostRuntime, DotnetConstants.InProc6HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (isVerbose)
+                {
+                    PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc6HostRuntime);
+                }
+                return true;
+            }
+            else
```

Comment:
nit: Remove the `else { ` and just keep the throw. (reduces some nesting and improve readability)